#!/usr/bin/env python
PACKAGE = "nezha_base"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("Kp",    int_t,    0, "Kp in PID", 1500,  0, 10000)
gen.add("Ki",    int_t,    0, "Ki in PID", 100,  0, 10000)
gen.add("Kd",    int_t,    0, "Kd in PID", 0,  0, 10000)

gen.add("ServoBias", int_t, 0, "ServoBias", 1500,  500, 2500)
gen.add("MaxSteeringAngle", double_t, 0, "MaxSteeringAngle", 0,  10, 28)
gen.add("SteeringOffsetFactor", double_t, 0, "SteeringOffsetFactor", 1.0,  0.0, 2.0)

exit(gen.generate(PACKAGE, "nezha_base", "pid_config"))
